src/widget/loginscreen.cpp

```diff
@@ -72,45 +73,56 @@ LoginScreen::~LoginScreen()
     delete ui;
 }
 
+void LoginScreen::closeEvent(QCloseEvent* event)
+{
+
+    // If we are in the bootstrap, returning -1 will give us something to exit with in main.cpp
+    this->setResult(-1);
+    // If we are in application exec, we can quit by closing it, instead.
+    qApp->quit();
```

Comment:
Because I do not believe it is possible at this time :/ 
In the first case we have called loginScreen as part of the initial startup, in which case we are about to call a->exec() (in main.cpp) and enter an empty application running indefinitely. The return value in that case will allow us to return early instead.

In the other case we have logged in (and thus are operating from the context of QApplication "a" in main.cpp) and then logged out. A closeEvent of LoginScreen is not sufficient at that stage to halt execution of the application, and nothing is listening to the return value either since showLogin was called by InvokeMethod from profileinfo.

I tried a few approaches to resolve this issue (Changing qApp->setQuitOnLastWindowClosed(false); to true, for one) but this was the only one I found that worked. I don't mean to say there couldn't be a better solution, but I could not find a better one.

Thank you for your other points, I will fix them ASAP. I also think I have a solution for the widget related mac OSX issue, again, at least to successfully build with some hands on testing recommended. Commits forthcoming, I have struggled to escape the event horizon of swedish midsummer celebrations.

Reply:
ok, this is an acceptable workaround then